pkg/hsperfdata/hsperfdata.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2022-2023 The Parca Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package hsperfdata
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/go-kit/log"
+
+	"github.com/parca-dev/parca-agent/pkg/perf"
+)
+
+const hsperfdata = "/tmp/hsperfdata_*"
+
+type Cache struct {
+	pids   map[int]bool
+	fs     fs.FS
+	logger log.Logger
+	mu     sync.Mutex
+	nsPID  map[int]int
+}
+
+type realfs struct{}
+
+func (f *realfs) Open(name string) (fs.File, error) {
+	return os.Open(name)
+}
+
+func NewCache(logger log.Logger) *Cache {
+	return &Cache{
+		pids:   make(map[int]bool),
```

Comment:
```suggestion
		pids:   make(map[int]struct{}),
```
⛏ nitpick: We can reduce the memory food print just by using an empty struct to represent the existence of the pid.